sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -729,6 +732,13 @@ public void validate() {
     private Instant curTimestamp;
     private Iterator<PartitionState> curBatch = Collections.emptyIterator();
 
+    private final Counter messagesConsumed = Metrics.counter("io", "messagesConsumed");
+    private final Counter bytesConsumed = Metrics.counter("io", "bytesConsumed");
```

Comment:
Alternately merge this PR with a few basic metrics like you have done here (you could probably skip 'per split' metrics), with a TODO to fix them up once a consensus is reached about the convention.

I don't see a reason why each reader shouldn't report basic metrics like `elementsRead` `backlog` since runner consumes those already. Simpler for each runner to report those metrics and an unbounded source could add more (e.g. bytes and other metrics).